lib/dat-worker-pool.rb

```diff
@@ -8,17 +8,22 @@
 
 class DatWorkerPool
 
-  attr_reader :logger, :queue
+  attr_reader :num_workers, :logger
+  attr_reader :queue
   attr_reader :on_worker_error_callbacks
   attr_reader :on_worker_start_callbacks, :on_worker_shutdown_callbacks
   attr_reader :on_worker_sleep_callbacks, :on_worker_wakeup_callbacks
   attr_reader :before_work_callbacks, :after_work_callbacks
 
-  def initialize(num_workers = 1, debug = false, &do_work_proc)
-    @num_workers  = num_workers
-    @debug        = debug
-    @logger       = Logger.new(@debug)
-    @do_work_proc = do_work_proc
+  def initialize(options = nil)
+    options ||= {}
+    @do_work_proc = options[:do_work_proc]
+    @num_workers  = (options[:num_workers] || 1).to_i
+    @logger       = options[:logger] || NullLogger.new
+
+    if @num_workers <= 0
```

Comment:
@jcredding maybe do a `MIN_WORKERS = 1` constant and change to `if @num_workers < MIN_WORKERS`.  Just to get the hard coded numbers moved to constants.